vignettes/Workflow.Rmd

```diff
@@ -0,0 +1,495 @@
+of the short read by a fixed amount in the appropriate direction,
+depending on whether the alignment was to the positive or negative
+strand, makes it more likely that we cover the actual site of
+interest.
+
+
+It is possible to estimate the fragment length, through a variety of
+methods. There are several implemented by the
+`estimate.mean.fraglen` function. Generally, this only needs
+to be done for one sample from each experimental protocol. Here, we
+use SSISR, the default method: 
+
+```{r estimate.mean.fraglen}
+fraglen <- estimate.mean.fraglen(cstest$ctcf, method="correlation")
+fraglen[!is.na(fraglen)] 
+```
+
+Given the suggestion of $~190$ nucleotides,
+we extend all reads to be 200 bases long. This is done using the
+`resize` function, which considers the strand to determine the direction
+of extension: 
+
+```{r}
+ctcf.ext <- resize (cstest$ctcf, width = 200)
+ctcf.ext
+```
+
+
+We now have intervals for the CTCF lane that represent the original
+fragments that were precipitated.
+
+# Coverage, islands, and depth
+
+A useful summary of this information is the
+*coverage*, that is, how many times each base in the genome was
+covered by one of these intervals.
+
+```{r}
+cov.ctcf <- coverage(ctcf.ext)
+cov.ctcf
+```
+
+For efficiency, the result is stored in a run-length encoded form.
+
+The regions of interest are contiguous segments of non-zero coverage,
+also known as *islands*.
+
+```{r}
+islands <- slice(cov.ctcf, lower = 1)
+islands
+```
+
+For each island, we can compute the number of reads in the island, and
+the maximum coverage depth within that island.
+
+```{r}
+viewSums(islands)
+viewMaxs(islands)
+
+nread.tab <- table(viewSums(islands) / 200)
+depth.tab <- table(viewMaxs(islands))
+
+nread.tab[,1:10]
+depth.tab[,1:10]
+```
+
+# Processing multiple lanes
+
+Although data from one lane is often a natural analytical unit, we
+typically want to apply any procedure to all lanes. 
+Here is a simple summary function that
+computes the frequency distribution of the number of reads.
+
+```{r}
+islandReadSummary <- function(x)
+{
+    g <- resize(x, 200)
+    s <- slice(coverage(g), lower = 1)
+    tab <- table(viewSums(s) / 200)
+    df <- DataFrame(tab)
+    colnames(df) <- c("chromosome", "nread", "count")
+    df$nread <- as.integer(df$nread)
+    df
+}
+```
+
+Applying it to our test-case, we get
+
+```{r}
+head(islandReadSummary(cstest$ctcf)) 
+``` 
+
+We can now use it to summarize
+the full dataset, flattening the returned *DataFrameList* with the
+`stack` function. 
+
+```{r} 
+nread.islands <- DataFrameList(lapply(cstest, islandReadSummary)) 
+nread.islands <- stack(nread.islands, "sample") 
+nread.islands 
+```
+
+```{r}
+xyplot(log(count) ~  nread | sample, as.data.frame(nread.islands),
+       subset = (chromosome == "chr10" & nread <= 40), 
+       layout = c(1, 2), pch = 16, type = c("p", "g"))
+```
+
+```{r fig.height=10, echo=FALSE}
+plot(trellis.last.object()) 
+```
```

Comment:
I noticed that in this vignette, a lot of the plots show up twice--not because of anything you did; I think I would have written the same code blocks as 199-203 and 205-207 because that's close to the .Rnw. Maybe this is the difference between how Sweave and R Markdown handle plots in code blocks or at least the `xyplot`, which I am not familiar with.

Before you do anything, I am going to ask another core team member's opinion since I usually don't deviate from the .Rnw

I want to propose removing the `plot(trellis.last.object())` blocks since they are just replotting the last object. For example, delete lines 205-207 and at line 200, add `fig.height=10` as an option in the opening part of the code block such that it becomes

   ```{r fig.height=10}

We don't include `echo=FALSE` because we do want the code to show up. We'd do this for any plots that appear twice.

Reply:
I think we can proceed with removing the duplicates. There is one special one that should be modified differently and I will mark it on the .Rmd.